src/aspire/operators/filters.py

```diff
@@ -33,6 +33,30 @@ def wavelength_to_voltage(wavelength):
     ) / (2 * 0.978466)
 
 
+def evaluate_grid_src(src, L, power=1):
+    """
+    Given an ImageSource object, compute the source's unique filters
+    at the filter_indices specified in its metadata.
+    :return: an `L x L x len(src.filter_indices)` array containing the evaluated
+    filters at each gridpoint
+    """
+    grid2d = grid_2d(L, dtype=src.dtype)
+    omega = np.pi * np.vstack((grid2d["x"].flatten(), grid2d["y"].flatten()))
+
+    h = np.empty((omega.shape[-1], len(src.filter_indices)), dtype=src.dtype)
+    for i, filt in enumerate(src.unique_filters):
+        idx_k = np.where(src.filter_indices == i)[0]
+        if len(idx_k) > 0:
+            filter_values = filt.evaluate(omega)
+            if power != 1:
```

Comment:
While we're here, I would remove the `power` argument and simply have the calling functions square the filter values if needed. I don't think there's a performance loss here and it would simplify the code.